runners/core-java/src/main/java/org/apache/beam/runners/core/triggers/TriggerStateMachines.java

```diff
@@ -60,6 +65,14 @@ public static OnceTriggerStateMachine stateMachineForOnceTrigger(OnceTrigger tri
   @VisibleForTesting
   static class StateMachineConverter {
 
+    /**
+     * Annotation indicating that a method is called via dynamic dispatch, not
+     * unused.
+     */
+    @Retention(RetentionPolicy.SOURCE)
+    @Target(ElementType.METHOD)
+    private @interface DynamicDispatch {}
```

Comment:
First, added annotation with `RUNTIME` lifetime, hoping findbugs would be smart. That failed, but it still seemed informative so I left it. I don't feel strongly. I'll probably remove it for concision and perhaps later actually drive the API by the annotations.